pycbc/inference/option_utils.py

```diff
@@ -564,7 +565,8 @@ def results_from_cli(opts, load_samples=True, **kwargs):
         logging.info("Reading input file %s", input_file)
 
         # read input file
-        fp = InferenceFile(input_file, "r")
+        fp = InferenceFile(input_file, "r") if input_file.endswith(".hdf") \
+                 else InferenceCSVFile(input_file, "r")
```

Comment:
I made a request related to this in #1921. Since you need to figure out in multiple places what sort of class to use, write a function that does in option utils that figures out what sort of file type to load based on the file extension.